### PR TITLE
[Slider] Change `Slider.Output` to `Slider.Value` and add `format` prop

### PR DESCRIPTION
**Pull request by mj12albert**
Part of https://github.com/mui/base-ui/issues/961

- `Slider.Value` matches the rendering API of [`Select.Value`](https://master--base-ui.netlify.app/react/components/select#value)
- `format` matches `NumberField` and does not take an array for range sliders, as it would make `aria-valuetext` for range sliders inconsistent ([comment](https://github.com/mui/base-ui/issues/961#issuecomment-2527118439))

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/base-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/react/src/slider/value/SliderValue.tsx [hunk 1]
```diff
@@ -6,34 +6,49 @@ import { useComponentRenderer } from '../../utils/useComponentRenderer';
...
 /**
- * Displays the current value of the slider as text.
- * Renders an `<output>` element.
+ * Demos:
+ *
+ * * [Slider](https://base-ui.com/components/react-slider/)
+ *
+ * API:
+ *
+ * * [SliderValue API](https://base-ui.com/components/react-slider/#api-reference-SliderValue)
```
Comment (mnajdova) [inline packages/react/src/slider/value/SliderValue.tsx:17]: This shouldn't be here, it's using the old format.

**Review by mnajdova — COMMENTED** [review]
Can you please update the demo on the new docs page and ask Vlad for a review? The implementation itself looks good, but the comment I added above causes issues on the docs.

Screenshot 2024-12-11 at 19 30 16

File: docs/reference/overrides/slider-root.json [hunk 2]
```diff
@@ -1,6 +1,9 @@
 {
```
Comment (mj12albert) [inline docs/reference/overrides/slider-root.json:1]: @vladmoroz The `*.Value` components also support children as a function (with some props) for formatting: https://github.com/mui/base-ui/blob/master/docs/data/components/select/select.mdx?plain=1#L179-L183

Should we add them to `docs/reference/overrides` and displayed as well?

Comment (vladmoroz) [reply docs/reference/overrides/slider-root.json:1]: > Should we add them to docs/reference/overrides to show them in the props table as well?

Yes please!

**Review by vladmoroz — APPROVED** [review]

**Pull request merged**